File: 0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode *getIntersectionNode(struct ListNode *A, struct ListNode *B) {
    struct ListNode* a = A; struct ListNode* b = B;
    while(a != b) {
        a = a ? a->next : B;
        b = b ? b->next: A;
    }
    return a;
}

File: 0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} headA
 * @param {ListNode} headB
 * @return {ListNode}
 */
var getIntersectionNode = function(A, B) {
    let a = A; let b = B;
    while(a !== b) {
        a = a != null ? a.next : B;
        b = b != null ? b.next : A;
    }
    return a;
};